notebook/converted/benchmark_results.py

```diff
@@ -775,151 +1678,166 @@
+    new_df['Model'] = supervised_model
     
     new_df['seed'] = model_info[-1]
             
     tissue_metrics = pd.concat([tissue_metrics, new_df])
     
 tissue_metrics['train_count'] = tissue_metrics['train sample count']
-tissue_metrics['supervised'] = tissue_metrics['supervised'].str.replace('pytorch_supervised', 'three_layer_net')
+tissue_metrics['Model'] = tissue_metrics['Model'].str.replace('pytorch_supervised', 'three_layer_net')
 
 
-# In[52]:
+# In[105]:
 
 
 viz_df = tissue_metrics[tissue_metrics['is_pretrained'] == 'not_pretrained']
-plot = ggplot(viz_df, aes(x='train_count', y='balanced_accuracy', color='supervised'))
+plot = ggplot(viz_df, aes(x='train_count', y='balanced_accuracy', color='Model'))
 plot += geom_smooth()
 plot += geom_point(alpha=.2)
 plot += ggtitle('Study-based split, study corrected')
 plot
 
 
-# ## GTEx All Tissues
+# ## Mutation prediction
 
-# In[2]:
+# In[106]:
 
 
-in_files = glob.glob('../../results/gtex-all-tissue.*.tsv')
-in_files = [f for f in in_files if 'be_corrected' not in f]
+in_files = glob.glob('../../results/tcga-binary.*.tsv')
 print(in_files[:5])
 
 
-# In[3]:
+# In[107]:
 
 
 tissue_metrics = pd.DataFrame()
 for path in in_files:
     new_df = pd.read_csv(path, sep='\t')
-    model_info = path.strip('.tsv').split('all-tissue.')[-1]
+    model_info = path.strip('.tsv').split('tcga-binary.')[-1]
     model_info = model_info.split('_')
         
     supervised_model = '_'.join(model_info[:-1])
              
-    new_df['supervised'] = supervised_model
+    new_df['Model'] = supervised_model.split('.')[-1]
+    new_df['gene'] = supervised_model.split('.')[0]
     
     new_df['seed'] = model_info[-1]
         
     tissue_metrics = pd.concat([tissue_metrics, new_df])
     
 tissue_metrics['train_count'] = tissue_metrics['train sample count']
 
-tissue_metrics['supervised'] = tissue_metrics['supervised'].str.replace('pytorch_supervised', 'three_layer_net')
-tissue_metrics['supervised'] = tissue_metrics['supervised'].str.replace('deep_net', 'five_layer_net')
-tissue_metrics
+tissue_metrics['Model'] = tissue_metrics['Model'].str.replace('pytorch_supervised', 'three_layer_net')
+tissue_metrics['Model'] = tissue_metrics['Model'].str.replace('deep_net', 'five_layer_net')
 
 
-# In[6]:
+# In[108]:
 
 
-plot = ggplot(tissue_metrics, aes(x='train_count', y='balanced_accuracy', color='supervised')) 
+plot_df = tissue_metrics[tissue_metrics['Model'].isin(['three_layer_net', 'five_layer_net', 'pytorch_lr'])]
+plot = ggplot(plot_df, aes(x='train_count', y='balanced_accuracy', color='Model'))
+plot += geom_hline(yintercept=.5, linetype='dashed')
 plot += geom_smooth()
-plot += geom_point(alpha=.2)
-plot += ggtitle('All Tissue Prediction GTEx Data (Patient-level split)')
+#plot += geom_point(alpha=.2)
+plot += facet_wrap(['gene'])
+plot += ggtitle('TCGA mutation prediction')
+plot += xlab('Train sample count')
+plot += ylab('Balanced Accuracy (Val Set)')
+plot.save('../../figures/tcga.svg')
```

Comment:
Interesting! I'm surprised that the five-layer net does so much better in the case of EGFR/KRAS/SETD2. I've tried random forests for those genes and didn't see much of an improvement over elastic net, but there were a few differences (we looked at AUPR rather than balanced accuracy, we used some additional covariates, we did cancer type filtering which I'm not sure if you did, etc).

I'm planning to try a shallow-ish NN for our multi-omics problem, so maybe I'll try it for gene expression too and see if I get similar results to you here.

Your TP53 and IDH1 results are pretty similar to what I would have expected, IDH1 is an easy classification problem and TP53 seems to be easy up until a point.

Reply:
Huh interesting. I think our problems are a little bit different since I'm not splitting out cancer types, it may be an artifact of that. Alternatively it could be that a smooth decision boundary is at as important as nonlinearity which is why the random forest doesn't look great but a five layer network does (but in that case I'd expect the three layer network to also do very well).